zerver/lib/test_classes.py

```diff
@@ -167,6 +184,35 @@ def run(self, result: Optional[TestResult]=None) -> Optional[TestResult]:  # noc
     DEFAULT_SUBDOMAIN = "zulip"
     TOKENIZED_NOREPLY_REGEX = settings.TOKENIZED_NOREPLY_EMAIL_ADDRESS.format(token="[a-z0-9_]{24}")
 
+    @contextmanager
+    def assertNoLogs(self, test_case,logger_name, level, no_logs, exc_type, exc_value,tb):
+        LOGGING_FORMAT = "%(levelname)s:%(name)s:%(message)s"
+        formatter = logging.Formatter(LOGGING_FORMAT)
+        handler = _CapturingHandler()
+        handler.setLevel(level)
+        handler.setFormatter(formatter)
+        self.watcher = handler.watcher
+        self.old_handlers = logger.handlers[:]
+        self.old_level = logger.level
+        self.old_propagate = logger.propagate
+        logger.handlers = [handler]
```

Comment:
what is `logger` here? I think even if we correct above errors we would get `NameError: name 'logger' is not defined` upon running this.

Reply:
Please have a look at new solution posted.